### DISCUSSION TITLE
Datagrid: add filter for date ranges

### DISCUSSION THREAD

**Discussion by derkoe** [vmware-clarity/ng-clarity#227] [Ideas]
The Datagrid has built-in filters supporting string and number fields. It would be great to also include a filter for date (or even date-time) objects. Currently, we have a custom filter for that but I think the use case to filter dates is very common.

Here is how the UI would look like:
![image](https://user-images.githubusercontent.com/123199/177169507-00a70368-0214-43e3-98aa-eed46932212e.png)

**Comment by d-m-s** [comment]
@gnomeontherun, @hippee-lee could you guys give us some feedback on this?

**Reply by d-m-s** [reply]
regarding "**waiting:needs-design**"

some smart guys proposed a really neat ui-design some years ago ;-)
![2022-08-09_14h53_36](https://user-images.githubusercontent.com/311619/183652334-a9ad4fca-5a9f-4dbc-acaf-f9971cc31cbe.png)

https://github.com/vmware/clarity/issues/474

Can we speed things up, just a little?

**Reply by bbogdanov** [reply]
> @gnomeontherun, @hippee-lee could you guys give us some feedback on this?

Jeremy and Matt are not part of the team anymore. @strefethen

**Reply by strefethen** [reply]
Would you mind adding some more specifics? I looked at the linked issue and the last comment reads:

gnomeontherun closed this as completed on Feb 5, 2021

**Reply by d-m-s** [reply]
> Jeremy and Matt are not part of the team anymore. 

@bbogdanov: Thanks for the info, wasn't aware of that. 
Who is leading the project now? Who can be contacted for strategic vision and to align coworking ideas of the project?

@strefethen **As mentioned above:** 
For datagrids which have date-columns, an appropriate/probably built-in filter would be great.

Some usecases to be considered: 
- **Date**: select just single day.
- **DateRange**: select start and end date. How this can look like, has been describe by your team in [#474](https://github.com/vmware/clarity/issues/474)
- **DateTime**: select day and time of day.
- **DateTimeRange**: select start and end date as well as time.

For the beginning I'm sure Date would be sufficient, just get some consistent throughout large applications - with custom filters, theses "standard-usecases" are implemented always differently and hence harm the usability as well as the efficiency of Clarity for users as well as for developers.

**Reply by kevinbuhmann** [reply]
There is an [open PR](https://github.com/vmware-clarity/ng-clarity/pull/250) for this, but it is on hold due because it needs design. We cannot merge the PR as is because it introduces a nested popover.

**Comment by strefethen** [comment]
@d-m-s VMware recently hired a Clarity Product Manager, @pdaigle.

**Comment by jerone** [comment]
You might be interested in the [Daterangepicker from Clarity Addons](https://porscheinformatik.github.io/clarity-addons/documentation/latest/daterangepicker).

![image](https://github.com/vmware-clarity/ng-clarity/assets/55841/7b2501b6-121f-4339-a5c9-a78410f81a2d)

It is not a filter, but it could be re-used for that.
